lib/gui/pages/settings/controllers/settings.js

```diff
@@ -53,34 +53,45 @@ module.exports = function(WarningModalService, SettingsModel) {
   this.model = SettingsModel;
 
   /**
-   * @summary Enable a dangerous setting
+   * @summary Toggle setting
    * @function
    * @public
    *
-   * @param {String} name - setting name
-   * @param {Object} options - options
-   * @param {String} options.description - modal description
-   * @param {String} options.confirmationLabel - modal confirmation label
+   * @description
+   * If warningOption is given, it should be an object having `description` and `confirmationLabel`;
+   * these will be used to present a user confirmation modal before enabling the setting.
+   *
+   * @param {String} setting - setting key
+   * @param {Object} warningOptions - (optional) WarningModalService options
    * @returns {Undefined}
    *
    * @example
-   * SettingsController.enableDangerousSetting('unsafeMode', {
+   * SettingsController.toggle('unsafeMode', {
    *   description: 'Don\'t do this!',
    *   confirmationLabel: 'Do it!'
    * });
    */
-  this.enableDangerousSetting = (name, options) => {
-    if (!this.currentData[name]) {
-      this.model.set(name, false);
-      return this.refreshSettings();
+  this.toggle = (setting, warningOptions) => {
+
+    const value = this.currentData[setting];
+    const dangerous = Boolean(warningOptions);
```

Comment:
I think it's okay, because for any non-dangerous settings, `warningOptions` won't be supplied. Perhaps it would be better if the code was more explicit, like `const dangerous = (warningOptions == undefined);` (or whatever the correct JS / nodeJS syntax is ;) )